lib/utils/analytics.ts

```diff
@@ -19,23 +19,45 @@ function initiateAnalytics() {
  * @param {string} property - The property of the event
  * @param {string} value - The value of the event
  */
-function useAnalytics() {
-  const { user } = useSupabaseAuth();
+async function captureAnalytics({ title, property, value }: AnalyticEvent) {
+  const analyticsObject: Record<string, string> = {};
+  const {
```

Comment:
I have user info retrieval happening in `captureAnalytics` atm, but it doesn't feel right as it's not the responsibility of the `captureAnalytics` functions to get the user info. It should be passed in.

To pass it in though, I need to rework some things in the app.